crates/proof-of-sql/src/proof_primitive/dory/public_parameters.rs

```diff
@@ -50,4 +60,275 @@ impl PublicParameters {
             max_nu,
         }
     }
+    #[cfg(feature = "std")]
+    /// Function to save `PublicParameters` to a file in binary form
```

Comment:
@Dustin-Ray Do you expect to have people inspect the content of such files? If so text (i.e. strings) are better than binary.

Reply:
I do not expect users to visually inspect the contents of this file, the reason being that these will be filled with opaque cryptographic group elements that have little context or meaning from visual inspection on their own. Further, printing them as strings will yield the unwieldy string representations used by the arkff crate which usually is printed as an array of u64 values in montgomery form. To make any sense of these values, they must be converted first to standard form, then a string representation is needed to directly print the value. I am unaware of such a method string method is use anywhere in our code. I am aware of a method that we use to parse strings *into* group elements, but not necessarily the other way around.

Lastly, the overhead of string parsing large setups on the order of nu >= 20 also seems untenable in my opinion. It is already going to be rather expensive to generate these larger setups. Adding string parsing to this process will create an undue burden not only in terms of efficiency but also in terms of reliability. Given our time constraints and priorities, I do not have the bandwidth to implement string parsing and writing, as well as the checks involved with ensuring that process is correct.

For visual inspection, one can read in the file and deserialize it, then choose the best way to visually represent the group elements as they see fit in rust code.

If visual string based inspections is something that is heavily requested, it might be something we could create a bounty on. However, I am not aware of any major chain employing the use of a proof system such as Groth16 in zcash, for example, taking the time and effort to add string parsing into their public setups.